From 58996610c83a36722b4558876506235ec601f0f6 Mon Sep 17 00:00:00 2001
From: Jason Gerecke <killertofu@gmail.com>
Date: Thu, 9 Mar 2017 10:44:10 -0800
Subject: [PATCH libwacom 1/7] data: add multiple missing ISDv4 devices

Adds the following devices which are present in the wacom-hid-descriptors
database but not libwacom:

 * E2 (HP Pavillion dv6)
 * 104 (Cube i7 Stylus)
 * 10E (Fujitsu Lifebook T732)
 * 117 (Panasonic Toughbook CF-C2)
 * 124 (Fujitsu ARROWS Tab QH55/M)
 * 5013 (Toshiba dynabook KIRA L93/39M)
 * 5014 (Lenovo ThinkPad Helix 2)
 * 503E (Lenovo ThinkPad X1 Yoga)
 * 503F (Lenovo ThinkPad Yoga X1)
 * 5048 (Lenovo ThinkPad Yoga 260)
 * 5044 (Lenovo ThinkPad Yoga 260)

Signed-off-by: Jason Gerecke <jason.gerecke@wacom.com>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
(cherry picked from commit 3b37fe15814cfa4c0ed72722edb0225d18563c5c)
---
 data/isdv4-104.tablet  | 14 ++++++++++++++
 data/isdv4-10e.tablet  | 14 ++++++++++++++
 data/isdv4-117.tablet  | 14 ++++++++++++++
 data/isdv4-124.tablet  | 15 +++++++++++++++
 data/isdv4-5013.tablet | 14 ++++++++++++++
 data/isdv4-5014.tablet | 15 +++++++++++++++
 data/isdv4-503e.tablet | 16 ++++++++++++++++
 data/isdv4-503f.tablet | 16 ++++++++++++++++
 data/isdv4-5044.tablet | 17 +++++++++++++++++
 data/isdv4-5048.tablet | 16 ++++++++++++++++
 data/isdv4-e2.tablet   | 14 ++++++++++++++
 11 files changed, 165 insertions(+)
 create mode 100644 data/isdv4-104.tablet
 create mode 100644 data/isdv4-10e.tablet
 create mode 100644 data/isdv4-117.tablet
 create mode 100644 data/isdv4-124.tablet
 create mode 100644 data/isdv4-5013.tablet
 create mode 100644 data/isdv4-5014.tablet
 create mode 100644 data/isdv4-503e.tablet
 create mode 100644 data/isdv4-503f.tablet
 create mode 100644 data/isdv4-5044.tablet
 create mode 100644 data/isdv4-5048.tablet
 create mode 100644 data/isdv4-e2.tablet

diff --git a/data/isdv4-104.tablet b/data/isdv4-104.tablet
new file mode 100644
index 0000000..358212c
--- /dev/null
+++ b/data/isdv4-104.tablet
@@ -0,0 +1,14 @@
+# this is for the Wacom pen as found in the Cube i7 Stylus
+
+[Device]
+Name=Wacom ISDv4 104
+DeviceMatch=i2c:056a:0104
+Class=ISDV4
+Width=9
+Height=5
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=false
+Buttons=0
diff --git a/data/isdv4-10e.tablet b/data/isdv4-10e.tablet
new file mode 100644
index 0000000..796e69b
--- /dev/null
+++ b/data/isdv4-10e.tablet
@@ -0,0 +1,14 @@
+# this is for the Wacom pen + touchscreen as found in the Fujitsu Lifebook T732
+
+[Device]
+Name=Wacom ISDv4 10E
+DeviceMatch=usb:056a:010e
+Class=ISDV4
+Width=11
+Height=6
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-117.tablet b/data/isdv4-117.tablet
new file mode 100644
index 0000000..1aea2a1
--- /dev/null
+++ b/data/isdv4-117.tablet
@@ -0,0 +1,14 @@
+# this is for the Wacom pen + touchscreen as found in the Panasonic Toughbook CF-C2
+
+[Device]
+Name=Wacom ISDv4 117
+DeviceMatch=usb:056a:0117
+Class=ISDV4
+Width=11
+Height=6
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-124.tablet b/data/isdv4-124.tablet
new file mode 100644
index 0000000..8540eef
--- /dev/null
+++ b/data/isdv4-124.tablet
@@ -0,0 +1,15 @@
+# this is for the Wacom pen as found in the Fujitsu ARROWS Tab QH55/M
+# the touchscreen is 056a:5008
+
+[Device]
+Name=Wacom ISDv4 124
+DeviceMatch=i2c:056a:0124
+Class=ISDV4
+Width=9
+Height=5
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=false
+Buttons=0
diff --git a/data/isdv4-5013.tablet b/data/isdv4-5013.tablet
new file mode 100644
index 0000000..1f5211a
--- /dev/null
+++ b/data/isdv4-5013.tablet
@@ -0,0 +1,14 @@
+# Active electrostatic (AES) sensor used by the Toshiba dynabook KIRA L93/39M
+
+[Device]
+Name=Wacom ISDv4 5013
+DeviceMatch=i2c:056a:5013
+Class=ISDV4
+Width=12
+Height=7
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-5014.tablet b/data/isdv4-5014.tablet
new file mode 100644
index 0000000..8f26933
--- /dev/null
+++ b/data/isdv4-5014.tablet
@@ -0,0 +1,15 @@
+# Active electrostatic (AES) sensor used by some versions of the
+# Lenovo ThinkPad Helix 2
+
+[Device]
+Name=Wacom ISDv4 5014
+DeviceMatch=i2c:056a:5014
+Class=ISDV4
+Width=10
+Height=6
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-503e.tablet b/data/isdv4-503e.tablet
new file mode 100644
index 0000000..3bf501b
--- /dev/null
+++ b/data/isdv4-503e.tablet
@@ -0,0 +1,16 @@
+# this is for the Wacom pen + touchscreen
+# as found in some versions of the Lenovo
+# Thinkpad X1 Yoga.
+
+[Device]
+Name=Wacom ISDv4 503E
+DeviceMatch=usb:056a:503e
+Class=ISDV4
+Width=12
+Height=7
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-503f.tablet b/data/isdv4-503f.tablet
new file mode 100644
index 0000000..1594fc8
--- /dev/null
+++ b/data/isdv4-503f.tablet
@@ -0,0 +1,16 @@
+# this is for the Wacom pen + touchscreen
+# as found in some versions of the Lenovo
+# Thinkpad X1 Yoga.
+
+[Device]
+Name=Wacom ISDv4 503F
+DeviceMatch=usb:056a:503f
+Class=ISDV4
+Width=12
+Height=7
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-5044.tablet b/data/isdv4-5044.tablet
new file mode 100644
index 0000000..d7ec420
--- /dev/null
+++ b/data/isdv4-5044.tablet
@@ -0,0 +1,17 @@
+# this is for the Wacom pen + touchscreen
+# as found in the Lenovo ThinkpPad Yoga 260
+# 20FD002HUS (12.5" FHD; 276x155mm).
+# Pen with 2 buttons, no eraser.
+
+[Device]
+Name=Wacom ISDv4 5044
+DeviceMatch=usb:056a:5044
+Class=ISDV4
+Width=10
+Height=6
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-5048.tablet b/data/isdv4-5048.tablet
new file mode 100644
index 0000000..c0b2eaf
--- /dev/null
+++ b/data/isdv4-5048.tablet
@@ -0,0 +1,16 @@
+# this is for the Wacom pen + touchscreen
+# as found in some versions of the Lenovo
+# ThinkpPad Yoga 260
+
+[Device]
+Name=Wacom ISDv4 5048
+DeviceMatch=usb:056a:5048
+Class=ISDV4
+Width=10
+Height=6
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=true
+Buttons=0
diff --git a/data/isdv4-e2.tablet b/data/isdv4-e2.tablet
new file mode 100644
index 0000000..c6de76d
--- /dev/null
+++ b/data/isdv4-e2.tablet
@@ -0,0 +1,14 @@
+# this is for the Wacom pen found in the HP Pavillion dv6
+
+[Device]
+Name=Wacom ISDv4 E2
+DeviceMatch=usb:056a:00e2
+Class=ISDV4
+Width=14
+Height=8
+IntegratedIn=Display;System
+
+[Features]
+Stylus=true
+Touch=false
+Buttons=0
-- 
2.13.5

